### DISCUSSION TITLE
Add PactumJS contract testing to CICD pipeline

### DISCUSSION THREAD

**Discussion by orenhan** [pactumjs/pactum#335] [Q&A]
Our team wants to implement contract testing between our services, when researching for most compatible tool for us we encountered PactumJS. As we understood, consumer and provider are both uploading contracts to pactum flow server, it compares them and results are shown in the compatibility matrix. I wonder if there is an api or something to the pactum server we can use to get comparation results in our cicd and fail builds if contract is broken by one of the sides, haven't found any documentation about it yet.  
Thanks :)

**Comment by ASaiAnudeep** [comment]
Hello @orenhan , 

We can use the `checkQualityGate` property from the [pactum-flow-plugin](https://github.com/pactumjs/pactum-flow-plugin) to fail builds in case of contracts are broken.

For API Information, you can check the swagger docs - http://localhost:3000/api/flow/v1/  or [in pactum-flow-plugin](https://github.com/pactumjs/pactum-flow-plugin/blob/main/src/clients/flow.js)
